runner/update-status

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+set -Eeo pipefail
+if [[ ${1:-} == '' ]]; then
+  # shellcheck source=runner/logger.bash
+  source logger.bash
+  log.error "Missing required argument -- '<phase>'"
+  exit 64
+fi
+
+if [ "${RUNNER_STATUS_UPDATE_HOOK:-false}" == "true" ]; then
+
+    apiserver=https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT_HTTPS}
+    serviceaccount=/var/run/secrets/kubernetes.io/serviceaccount
+    namespace=$(cat ${serviceaccount}/namespace)
+    token=$(cat ${serviceaccount}/token)
+
+    jq -n --arg phase "$1" --arg message "$2" '.status.phase = $phase | .status.message = $message' | HTTPS_PROXY= http_proxy= curl \
+        --cacert ${serviceaccount}/ca.crt \
+        --data @- \
+        --header "Content-Type: application/merge-patch+json" \
+        --header "Authorization: Bearer ${token}" \
+        --show-error \
+        --silent \
+        --output /dev/null \
+        --request PATCH \
+        ${apiserver}/apis/actions.summerwind.dev/v1alpha1/namespaces/${namespace}/runners/${HOSTNAME}/status > /dev/null
```

Comment:
- If we want to disable proxies we should use `--noproxy '*'` that disables it for all hosts. However, I'm not sure if we actually want this, it should be a user choice.
- If we want curl to be truly silent then we can just close its stdout. (We should not ignore the error output, and we are not right now, curl failures will fail our script because we use `set -e` and this is how it should be.)
- We have to quote the URL to avoid issues with possible whitespace.

```suggestion
    jq -n --arg phase "$1" --arg message "$2" '.status.phase = $phase | .status.message = $message' | curl \
        --cacert ${serviceaccount}/ca.crt \
        --data @- \
        --header "Content-Type: application/merge-patch+json" \
        --header "Authorization: Bearer ${token}" \
        --noproxy '*' \
        --show-error \
        --silent \
        --request PATCH \
        "${apiserver}/apis/actions.summerwind.dev/v1alpha1/namespaces/${namespace}/runners/${HOSTNAME}/status"
        1>&-
```